contracts/src/PBHVerifier.sol

```diff
@@ -1,32 +1,176 @@
+    // these variables takes place. If reordering happens, a storage clash will occur (effectively a
+    // memory safety error).
+
+    using ByteHasher for bytes;
+
     ///////////////////////////////////////////////////////////////////////////////
-    ///                             CONSTRUCTION                                ///
+    ///                                  ERRORS                                ///
+    //////////////////////////////////////////////////////////////////////////////
+
+    /// @notice Thrown when attempting to reuse a nullifier
+    error InvalidNullifier();
+
     ///////////////////////////////////////////////////////////////////////////////
+    ///                                  Events                                ///
+    //////////////////////////////////////////////////////////////////////////////
 
-    /// @notice Constructs a new instance of the PBH Verifier.
-    /// @dev This constructor is only called once, and can be called with the encoded call necessary
-    ///      to initialize the logic contract.
+    /// @notice Emitted once for each successful PBH verification.
     ///
-    /// @param _logic The initial implementation (delegate) of the contract that this acts as a proxy
-    ///        for.
-    /// @param _data If this is non-empty, it is used as the data for a `delegatecall` to `_logic`.
-    ///        This is usually an encoded function call, and allows for initialising the storage of
-    ///        the proxy in a way similar to a traditional solidity constructor.
-    constructor(address _logic, bytes memory _data) payable WorldIDProxy(_logic, _data) {
-        // !!!! DO NOT PUT PROGRAM LOGIC HERE !!!!
-        // It should go in the `initialize` function of the delegate instead.
+    /// @param root The root of the Merkle tree that this proof is valid for.
+    /// @param sender The sender of this particular transaction or UserOp.
+    /// @param nonce Transaction/UserOp nonce.
+    /// @param callData Transaction/UserOp call data.
+    /// @param pbhExternalNullifier External nullifier encoding month, year, and a pbhNonce.
+    /// @param nullifierHash Nullifier hash for this semaphore proof.
+    /// @param proof The zero-knowledge proof that demonstrates the claimer is registered with World ID.
+    event PBH(
+        uint256 indexed root,
+        address indexed sender,
+        uint256 nonce,
+        bytes callData,
+        uint256 indexed pbhExternalNullifier,
+        uint256 nullifierHash,
+        uint256[8] proof
+    );
+
+    ///////////////////////////////////////////////////////////////////////////////
+    ///                                  Vars                                  ///
+    //////////////////////////////////////////////////////////////////////////////
+
+    /// @dev Gap for transient storage.
+    uint256 private __gap;
+
+    /// @dev The World ID group ID (always 1)
+    uint256 internal immutable _GROUP_ID = 1;
```

Comment:
Can we set this as a constant if we are assigning the value here? Or is there value as keeping it as immutable?